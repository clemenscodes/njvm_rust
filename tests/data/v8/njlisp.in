(PLUS 3 4)
(TIMES (PLUS 3 4) (DIFFERENCE 10 7))
(PUTD (QUOTE SQUARE) (QUOTE (LAMBDA (X) (TIMES X X))))
(SQUARE 12345)
(SQUARE (SQUARE (SQUARE 12345)))
(OBLIST)
123456789*987654321;
1-2*3+4*5;
FUNCTION F(N), WHEN N=0 OR N=1, N EXIT, F(N-1)+F(N-2) ENDFUN;
F(10);
FUNCTION G(N), A:0, B:1, LOOP WHEN N=0, A EXIT, H:A+B, A:B, B:H, N:N-1 ENDLOOP ENDFUN;
G(100);
5/9 + 7/12;
((236 - 3*127) * -13) ^ 16;
GCD(861, 1827);
(-24) ^ (1/3);
(-4) ^ (1/2);
#E ^ (3 * #I * #PI / 2);
5*X^2/X - 3*X^1;
(5*X)^3 / X;
EXPD((3*Y^2 - 2*Y + 5)^3);
FCTR(6*X^2*Y - 4*X*Y^2/Z);
(SYSTEM)