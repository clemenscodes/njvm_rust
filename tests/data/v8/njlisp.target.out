Ninja Virtual Machine started

Welcome to Ninja LISP!

$
7

$
21

$
(LAMBDA (X) (TIMES X X))

$
152399025

$
539422123247359763587428687890625

$
(X SQUARE PURGENAME XCHGPNAME PUSH POP SETQ OR AND PROG1 LOOP COND QUOTE LIST REMAINDER QUOTIENT DIVIDE TIMES DIFFERENCE PLUS MINUS EVEN ZEROP MINUSP PLUSP LESSP GREATERP NUMBERP SYSTEM SAVE LOAD RADIX LINELENGTH SPACES TERPRI RECLAIM EVAL APPLY EQUAL EQ ORDERED ORDERP NOT NULL NAME ATOM OBLIST REVERSE LENGTH UNPACK PACK GETD PUTD MOVD REMFLAG FLAG FLAGP REMPROP MEMBER GET PUT ASSOC SET NCONC RPLACD RPLACA CONS CDADR CAADR CDDAR CADAR CDAAR CAAAR CDAR CAAR CDDDR CADDR CDDR CADR CDR CAR DRIVER WRS PRIN1 PRINT RDS READCH RATOM READ ECHO NLAMBDA LAMBDA NSUBR SUBR T NIL)

$
123456789*987654321;

$
1-2*3+4*5;

$
FUNCTION

$
F

$
(N)

$
WHEN

$
N=0

$
OR

$
N=1

$
N

$
EXIT

$
F

$
(N-1)

$
+F

$
(N-2)

$
ENDFUN;

$
F

$
(10)

$
;

$
FUNCTION

$
G

$
(N)

$
A:0

$
B:1

$
LOOP

$
WHEN

$
N=0

$
A

$
EXIT

$
H:A+B

$
A:B

$
B:H

$
N:N-1

$
ENDLOOP

$
ENDFUN;

$
G

$
(100)

$
;

$
5/9

$
+

$
7/12;

$
((236 - 3*127) * -13)

$
^

$
16;

$
GCD

$
(861 1827)

$
;

$
(-24)

$
^

$
(1/3)

$
;

$
(-4)

$
^

$
(1/2)

$
;

$
#E

$
^

$
(3 * #I * #PI / 2)

$
;

$
5*X^2/X

$
-

$
3*X^1;

$
(5*X)

$
^3

$
/

$
X;

$
EXPD

$
((3*Y^2 - 2*Y + 5) ^3)

$
;

$
FCTR

$
(6*X^2*Y - 4*X*Y^2/Z)

$
;

$

Thanks for using Ninja LISP!

Ninja Virtual Machine stopped
